javascript/map.js

```diff
@@ -0,0 +1,49 @@
+const map = new maplibregl.Map({
+  container: 'map',
+  style: 'https://demotiles.maplibre.org/style.json', // stylesheet location
+  center: [-74.5, 40], // starting position [lng, lat]
+  zoom: 1 // starting zoom
+});
+
+map.addControl(new maplibregl.NavigationControl({ showCompass: false }));
+
+map.on('load', function() {
+  map.addSource('basemap-satellite', {
+    'type': 'raster',
+    "tiles": ["https://server.arcgisonline.com/ArcGIS/rest/services/World_Imagery/MapServer/tile/{z}/{y}/{x}"],
+  });
```

Comment:
You need to specify the tile size if you don't want the tiles to appear blurry:
```suggestion
  map.addSource('basemap-satellite', {
    type: 'raster',
    tiles: ['https://server.arcgisonline.com/ArcGIS/rest/services/World_Imagery/MapServer/tile/{z}/{y}/{x}'],
    tileSize: 256
  });
```